### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by ilzeLourens**

### CODE REVIEW

File: skietbaan_fe/src/components/Documents.js [hunk 1]
```diff
@@ -1,102 +1,71 @@
...
+          value: "",
+          value2: "",
+        };
+        this.SendLOGS = this.SendLOGS.bind(this)
+        this.SendLOS = this.SendLOS.bind(this)
+      }
+
+    componentWillMount(){
+        let token =getCookie("token");
+        fetch("http://api.skietbaan.retrotest.co.za/api/Documents/UserLOGS/" + token)
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/Documents.js]: BASE_URLS for all instances in this file

File: skietbaan_fe/src/components/Documents.js [hunk 2]
```diff
@@ -1,102 +1,71 @@
...
                         </div>
-                        <br />
-                    </div>
-                    <div className="button-upload-document">
-                        <Button className="send-email">SendEmail</Button>
-                    </div>
-                    <br />
+                                
+                        <div className="button-upload-document-3">
+                            <button className={this.state.value=="Document"? "btn-active send-email btn-bottom-3":"btn-default send-email btn-bottom-3"} onClick={this.state.value=="Document"? this.SendLOGS: null}> Letter of Good Standing {this.state.value=="Document"? <img src={require("../resources/sendDoc.png")}/>:null}</button>
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/Documents.js]: Split over multiple lines and images are supposed to be imported into variables 
Same for line 53 61 and 65

File: skietbaan_fe/src/components/Groups.js [hunk 3]
```diff
@@ -1,117 +1,179 @@
...
+          <button
+            className={this.state.count == 0 ? "select" : "select2"}
+            id="check"
+            onClick={this.selectall}
+          >
+            {this.state.check}
+          </button>
         </div>
+
+        <div className="OnToTheNextOne" />
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/Groups.js]: Inconsistent class names choose camelCase or CapitalCase. I suggest camelCase. This needs to reflect in all class names

File: skietbaan_fe/src/components/GroupsName.js [hunk 4]
```diff
@@ -25,7 +24,7 @@ onSubmit(){
         "Name": this.state.name,
       }
       localStorage.setItem(this.state.name,this.state.name);
-      fetch("http://localhost:63474/api/groups", {
+      fetch("https://api.skietbaan.retrotest.co.za/api/groups", {
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/GroupsName.js]: BASE_URL and empty line EOF

File: skietbaan_fe/src/components/Login.js [hunk 5]
```diff
@@ -114,64 +157,84 @@ class Login extends Component {
...
       window.location = "/home";
     }
 
     return (
-      <div className="Page-content">
-      <div className = "welcome-header"><label className="welcome-label">Welcome back to skietbaan</label></div>
-      <div className="header-container">
-      <label className = "header-label">Login</label>
-      </div>
+      <div className="page-content-login">
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/Login.js]: inconsistent class names. Take note of earlier comments. Choose between camelCase or dash-notation but not CapitalCase

File: skietbaan_fe/src/components/NavbarMenu.js [hunk 6]
```diff
@@ -30,11 +43,11 @@ class NavbarMenu extends Component {
 
 	isHome() {
 		if (window.location.pathname.endsWith("/home")) {
-			return <img src={require('../components/navbar-icons/leaderboard-red.png')}
+			return <img src={leaderboardRed}
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/NavbarMenu.js]: Inconsistent indentations in file `skietbaan_fe/src/components/Login.js` you used single indentation here it is not I would suggest to stick to single indentation

File: skietbaan_fe/src/components/NavbarMenuUser.js [hunk 7]
```diff
@@ -28,61 +38,83 @@ class NavbarMenuUser extends Component {
 
 	isHome() {
 		if (window.location.pathname.endsWith("/home")) {
-			return <img src={require('../components/navbar-icons/leaderboard-red.png')}
+			return <img src={leaderboardRed}
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/NavbarMenuUser.js]: Same note about indentation from above

File: skietbaan_fe/src/components/Notifications.js [hunk 8]
```diff
@@ -2,42 +2,48 @@ import React, { Component } from "react";
...
 
-  onDelete = key => {
-    let id = this.state.array[key].id;
-    let newArray = this.state.array;
-    newArray.splice(key, 1);
-    this.setState({ array: newArray });
+  onDelete = async () => {
+    const deleteNotification = async id => {
+      try {
+        await fetch(LOCALURL + `/api/Notification/${id}`, {
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/Notifications.js]: Is LOCALURL not a localhost url? Also all cap constants' words should be separated by '_' like LOCAL_URL

File: skietbaan_fe/src/components/Notifications.js [hunk 9]
```diff
@@ -52,47 +58,71 @@ class notification extends Component {
...
+  onClick_cancel() {
+    window.location = "/notify";
+  }
+
   componentDidMount() {
     if (getCookie("token")) {
-      var token = document.cookie;
-      fetch( URL + "/api/Notification?" + token)
+      const token = document.cookie;
+      fetch(LOCALURL + "/api/Notification?" + token)
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/Notifications.js]: again LOCAL_URL

File: skietbaan_fe/src/components/Register.js [hunk 10]
```diff
@@ -14,14 +20,31 @@ class Register extends Component {
...
+        'Accept': 'application/json',
+        'Content-Type': 'application/json'
+      }
+    })
+      .then(response => response.json())
+      .then(data => this.setState({
+        users: data,
+      }))
+      .catch(function (data) {
+        console.log("error")
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/Register.js]: remove log if it is for debug

File: skietbaan_fe/src/components/RegisterMember.js [hunk 11]
```diff
@@ -111,6 +114,8 @@ class App extends Component {
       })
       .then(function (data) { })
       .catch(function (data) { });
+
+      window.location = "/viewmembers";
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/RegisterMember.js]: All other paths were camelCase

File: skietbaan_fe/src/components/RegisterMember.js [hunk 12]
```diff
@@ -150,7 +155,7 @@ class App extends Component {
         <tbody className="table-body-create-members">
           {this.state.arrayUsers.map((post, index) => (
             <tr className="register-member-user-column" key={post.id} onClick={() => this.SearchMember(index)}>
-              <td className={this.state.clicked ? "actives" : "register-member-user-row"} onClick={() => this.ChangeColor()}>
+              <td className={this.state.clicked ? "actives" : "register-member-user-column"} onClick={() => this.ChangeColor()}>
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/RegisterMember.js]: split over multiple lines

File: skietbaan_fe/src/components/RegisterMemberStyles.css [hunk 13]
```diff
@@ -156,16 +164,19 @@
   }
   
   .register-member-user-column{
-    background-color: white;
+    background-color: #798CA5;
+    color: white;
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/RegisterMemberStyles.css]: You are using hex everywhere else please be consistent

File: skietbaan_fe/src/components/RegisterStyles.css [hunk 14]
```diff
@@ -177,20 +177,55 @@
...
+.back-btn{
+  float: left;
+  margin-top: -38px;
+  margin-left: 30px;
+  height: 34px;
+}
+
+.centre-label{
+  text-align: center;
+}
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/RegisterStyles.css]: empty line EOF

File: skietbaan_fe/src/components/ScoreCapture.js [hunk 15]
```diff
@@ -26,8 +30,10 @@ export default class search extends Component {
       Flashon: false,
       validScore: true,
       validCompetition: true,
-      scoreEntered:false
+      scoreEntered:false,
+      iconhouseUrl : '../components/assets/grayFlashOff.png'
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/ScoreCapture.js]: Other icons were imported please do the same here

File: skietbaan_fe/src/components/ScoreCapture.js [hunk 16]
```diff
@@ -69,6 +76,7 @@ export default class search extends Component {
   }
 
   componentDidMount() {
+    console.log(cameraGray)
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/ScoreCapture.js]: remove debug log

File: skietbaan_fe/src/components/ScoreCapture.js [hunk 17]
```diff
@@ -312,25 +335,26 @@ export default class search extends Component {
...
+        competitionItem.push(<div className="competition-item-container" key={'mykey' + i}> <div key={'mykey' + i}
           className={this.state.clicked != null && this.state.clicked === i ? "active competition-item" : "competition-item"}>
           <li onClick={() => this.CompetitionClicked(i, this.state.competitionsList[i].name)}>
             {this.state.competitionsList[i].name} </li></div></div>);
 
       }
     }
-
+    if(!getCookie("token")){
+      window.location = "/register-page";
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/ScoreCapture.js]: Other paths were camelCase please be consistent

File: skietbaan_fe/src/components/ViewGroups.css [hunk 18]
```diff
@@ -0,0 +1,70 @@
+.Top-bar {
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/ViewGroups.css]: With dash notation everything is lowercase

File: skietbaan_fe/src/components/ViewGroups.css [hunk 19]
```diff
@@ -0,0 +1,70 @@
...
+  float: right;
+  margin-top: 20px;
+}
+
+.blabe{
+    font-Size:"16px";
+    color: #fdfdfd;
+}
+
+.c{
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/ViewGroups.css]: Make use of meaningful class names

Comment (FredJEhlers) [reply skietbaan_fe/src/components/ViewGroups.css]: same for `blabe`

File: skietbaan_fe/src/components/ViewGroups.css [hunk 20]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+.theone{
+  width: 100%;
+  font-weight: lighter;
+  font-size: 18px;
+  color: #fdfdfd;
+}
+
+.theList{
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/ViewGroups.css]: again with inconsistent naming convention

File: skietbaan_fe/src/components/ViewGroups.css [hunk 21]
```diff
@@ -0,0 +1,70 @@
...
+  font-size: 18px;
+}
+
+.groupNames{
+  font-size: 18px;
+  font-weight: lighter;
+}
+.bpanel{
+  position: fixed;
+}
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/ViewGroups.css]: empty line EOF

File: skietbaan_fe/src/components/ViewGroups.js [hunk 22]
```diff
@@ -0,0 +1,163 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import "./groups.css";
+import history from "./history";
+import { passId } from "../actions/postActions";
+import { getname } from "../actions/postActions";
+import { BASE_URL } from "../actions/types";
+class ViewGroups extends Component {
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/ViewGroups.js]: Space between import and class

File: skietbaan_fe/src/components/ViewGroups.js [hunk 23]
```diff
@@ -0,0 +1,163 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import "./groups.css";
+import history from "./history";
+import { passId } from "../actions/postActions";
+import { getname } from "../actions/postActions";
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/ViewGroups.js]: Use naming convention consistently

File: skietbaan_fe/src/components/ViewGroups.js [hunk 24]
```diff
@@ -0,0 +1,163 @@
...
+      ShowMe: true,
+      ids: 0,
+      index: 0
+    };
+    this.onBack = this.onBack.bind(this);
+    this.onChange = this.onChange.bind(this);
+    this.delete = this.delete.bind(this);
+    this.editGroup = this.editGroup.bind(this);
+  }
+  UNSAFE_componentWillMount() {
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/ViewGroups.js]: Spaces between functions, No need for UNSAFE to be all caps

File: skietbaan_fe/src/components/ViewGroups.js [hunk 25]
```diff
@@ -0,0 +1,163 @@
...
+                  .toLowerCase()
+                  .startsWith(this.state.filterText.toLowerCase()) ||
+                post.email
+                  .toLowerCase()
+                  .startsWith(this.state.filterText.toLowerCase())
+              );
+            })
+            .map((post, index) => (
+              <li
+                style={{ width: "100%", background: "#504F51" }}
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/ViewGroups.js]: No inline styles

File: skietbaan_fe/src/components/ViewGroups.js [hunk 26]
```diff
@@ -0,0 +1,163 @@
...
+                <label className="the-container">
+                  <label
+                    className="nn"
+                    onClick={() => this.editGroup(post.id, post.name)}
+                  >
+                    <div className="groupNames"> {post.name}</div>
+                  </label>
+                  <div className="im">
+                    <img
+                      src={require("./GroupImages/submit plus add score.png")}
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/ViewGroups.js]: Import images into a variable

File: skietbaan_fe/src/components/ViewMembers.css [hunk 27]
```diff
@@ -14,31 +14,35 @@
   .centre-view-member{
     margin:0 auto;
     position: center;
+    background: white;
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/ViewMembers.css]: Use hex everywhere

File: skietbaan_fe/src/components/ViewMembers.css [hunk 28]
```diff
@@ -53,35 +57,40 @@
...
+    font-size: 18px;
+    color: #798CA5;
+    border: 1.5px #798CA5 solid;
   }
 
   .search {
-    padding-left: 45px;
-    padding-right: 45px;
-    padding-top: 10px;
+    padding-left: 30px;
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/ViewMembers.css]: Use shorthand notation for the padding

File: skietbaan_fe/src/components/ViewMembers.css [hunk 29]
```diff
@@ -90,29 +99,32 @@
...
   .view-members-user::selection{
-    background-color: #bbbcbd;
+    background-color: #6F839E;
+    color:white;
   }
 
   .view-members-user:hover{
-    background-color: lightgray;
+    background-color: #6F839E;
+    color:white;
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/ViewMembers.css]: Use hex

File: skietbaan_fe/src/components/ViewMembers.js [hunk 30]
```diff
@@ -109,6 +109,8 @@ class ViewMembers extends Component {
                             <td className="first-column">
                                 <Collapsible trigger={<div className="username-and-email">
                                     <b>{post.username}</b>
+                                    <img src={require('../components/assets/membership-icon.png')}
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/ViewMembers.js]: Import image into var

File: skietbaan_fe/src/components/add.css [hunk 31]
```diff
@@ -0,0 +1,98 @@
+.top_bar {
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/add.css]: dash notation not underscore

Comment (FredJEhlers) [reply skietbaan_fe/src/components/add.css]: and white => Hex

File: skietbaan_fe/src/components/add.css [hunk 32]
```diff
@@ -0,0 +1,98 @@
...
+  background: white;
+  height: 50px;
+  display: flex;
+  background: black;
+}
+
+.center_label {
+  display: flex;
+  margin-left: auto;
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/add.css]: Double declaration

File: skietbaan_fe/src/components/add.css [hunk 33]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+.texts {
+  border-radius: 10px;
+  height: 40px;
+  padding-left: 5px;
+  margin-bottom: 15px;
+  font-weight: lighter;
+}
+.page {
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/add.css]: Space between declarations

File: skietbaan_fe/src/components/add.css [hunk 34]
```diff
@@ -0,0 +1,98 @@
...
+  height: -webkit-fill-available;
+}
+
+.name {
+  color: white;
+  font-weight: lighter;
+  font-size: 18px;
+}
+.add {
+
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/add.css]: remove space

File: skietbaan_fe/src/components/groups.css [hunk 35]
```diff
@@ -1,85 +1,279 @@
...
+  float: right;
+  margin-top: 20px;
+  margin-right: 39px;
+  height: 17px;
+  width: 17px;
+  margin-top: 37px;
+  margin-bottom: 37px;
+}
+
+.Check {
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/groups.css]: Naming conventions

File: skietbaan_fe/src/components/groups.css [hunk 36]
```diff
@@ -1,85 +1,279 @@
...
-    float: right;
-    font-size: 20px;
-    color: black;
+
+.btplane{
+  display: flex;
 }
 
-.middle .add{
+.ubutns{
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/groups.css]: Use meaningful names

File: skietbaan_fe/src/components/groups.css [hunk 37]
```diff
@@ -1,85 +1,279 @@
...
+  font-weight: bold;
+  padding-left: 15px;
+  
+}
+.thetextname{
+  font-size: 16px;
+  padding-right: 30px;
+  font-weight: lighter;
+}
+.thetxt{
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/groups.css]: naming convention reminder also empty line EOF

File: skietbaan_fe/src/components/history.js [hunk 38]
```diff
@@ -1,2 +1,7 @@
-import { createBrowserHistory } from 'history';
+
+
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/history.js]: remove unneeded White space

File: skietbaan_fe/src/reducers/postReducer.js [hunk 39]
```diff
@@ -1,48 +1,75 @@
-import { FETCH_POSTS, NEW_POST,FETCH_LEADERBOARDFILTER_DATA ,FETCH_LEADERBOARDTABLE_DATA } from '../actions/types';
+import {
+  FETCH_POSTS,
+  UPDATE_GROUPNAME,
+  FETCH_LEADERBOARDFILTER_DATA,
+  FETCH_LEADERBOARDTABLE_DATA,
+  PASS_ID,
+  CREATEGROUP
+} from "../actions/types";
 const initialState = {
```
Comment (FredJEhlers) [inline skietbaan_fe/src/reducers/postReducer.js]: empty line between these declarations

File: skietbaan_fe/src/store.js [hunk 40]
```diff
@@ -1,16 +1,18 @@
...
 const initialState = {};
 const middleware = [thunk];
 const store = createStore(
-    rootReducer,
-    initialState,
-    compose( applyMiddleware(...middleware)));
+  rootReducer,
+  composeWithDevTools(applyMiddleware(...middleware))
+);
+
```
Comment (FredJEhlers) [inline skietbaan_fe/src/store.js]: no need for all this space

Comment (DinolanNaidoo) [reply skietbaan_fe/src/store.js]: removed spaces

File: skietbaan_fe/src/reducers/postReducer.js (hunk 39 above)
Comment (DinolanNaidoo) [reply skietbaan_fe/src/reducers/postReducer.js]: added empty lines

File: skietbaan_fe/src/components/history.js (hunk 38 above)
Comment (DinolanNaidoo) [reply skietbaan_fe/src/components/history.js]: removed

File: skietbaan_fe/src/components/NavbarMenu.js (hunk 6 above)
Comment (DinolanNaidoo) [reply skietbaan_fe/src/components/NavbarMenu.js]: made it consisitent

File: skietbaan_fe/src/components/NavbarMenuUser.js (hunk 7 above)
Comment (DinolanNaidoo) [reply skietbaan_fe/src/components/NavbarMenuUser.js]: made consistent

File: skietbaan_fe/src/components/RegisterMember.js (hunk 11 above)
Comment (DinolanNaidoo) [reply skietbaan_fe/src/components/RegisterMember.js]: sorted

File: skietbaan_fe/src/components/RegisterMember.js (hunk 12 above)
Comment (DinolanNaidoo) [reply skietbaan_fe/src/components/RegisterMember.js]: done

File: skietbaan_fe/src/components/RegisterMemberStyles.css (hunk 13 above)
Comment (DinolanNaidoo) [reply skietbaan_fe/src/components/RegisterMemberStyles.css]: done

File: skietbaan_fe/src/components/ViewMembers.css (hunk 27 above)
Comment (DinolanNaidoo) [reply skietbaan_fe/src/components/ViewMembers.css]: done

File: skietbaan_fe/src/components/ViewMembers.css (hunk 29 above)
Comment (DinolanNaidoo) [reply skietbaan_fe/src/components/ViewMembers.css]: done

File: skietbaan_fe/src/components/Login.js (hunk 5 above)
Comment (ilzeLourens) [reply skietbaan_fe/src/components/Login.js]: done

File: skietbaan_fe/src/components/ViewMembers.js (hunk 30 above)
Comment (DinolanNaidoo) [reply skietbaan_fe/src/components/ViewMembers.js]: done

File: skietbaan_fe/src/components/ScoreCapture.js (hunk 15 above)
Comment (ilzeLourens) [reply skietbaan_fe/src/components/ScoreCapture.js]: removed it

File: skietbaan_fe/src/components/ScoreCapture.js (hunk 16 above)
Comment (ilzeLourens) [reply skietbaan_fe/src/components/ScoreCapture.js]: removed

File: skietbaan_fe/src/components/ScoreCapture.js (hunk 17 above)
Comment (ilzeLourens) [reply skietbaan_fe/src/components/ScoreCapture.js]: changed path name

File: skietbaan_fe/src/components/ViewMembers.css (hunk 28 above)
Comment (DinolanNaidoo) [reply skietbaan_fe/src/components/ViewMembers.css]: done

File: skietbaan_fe/src/components/ViewGroups.js (hunk 24 above)
Comment (Bathande) [reply skietbaan_fe/src/components/ViewGroups.js]: this is a predefined function it only works when unsafe is in caps

**Pull request merged**